mbedtls-sys/build/cmake.rs

```diff
@@ -46,25 +46,20 @@ impl super::BuildConfig {
             cmk.define("CMAKE_TRY_COMPILE_TARGET_TYPE", "STATIC_LIBRARY");
         }
 
-        let mut dst = cmk.build();
+        let dst = cmk.build();
 
-        dst.push("lib");
         println!(
             "cargo:rustc-link-search=native={}",
-            dst.to_str().expect("link-search UTF-8 error")
+            dst.join("lib").to_str().expect("link-search UTF-8 error")
         );
 
-        println!("cargo:rustc-link-lib=mbedtls");
-        println!("cargo:rustc-link-lib=mbedx509");
-        println!("cargo:rustc-link-lib=mbedcrypto");
+        println!("cargo:rustc-link-lib=static=mbedtls");
+        println!("cargo:rustc-link-lib=static=mbedx509");
+        println!("cargo:rustc-link-lib=static=mbedcrypto");
```

Comment:
What's the affect of newly added `static=`? 
Does it just explicitly ensure static link?

Reply:
> What's the affect of newly added `static=`?
> Does it just explicitly ensure static link?

It does not have an effect in Cargo but setting static linking explicitly makes Bazel's Rust rule work properly.

More Context:
Since they are built script's output in a lib (rlib) crate, it should and would linked statically to the final ELF and Cargo handles it automatically even if we don't mention it.
But Rust's rule in Bazel is trusting what we asked for. Since the default value for linking is dynamic. it tries to find these libraries system defined shared libraries and causes failure.
[link the generated crate to a native library](https://doc.rust-lang.org/rustc/command-line-arguments.html#-l-link-the-generated-crate-to-a-native-library)

So it would be better for both to define explicitly.

Reply:
@irajtaghlidi why doesn't this change have an effect when building with Cargo?

On the page you linked above, I see

> The kind of library and the modifiers can also be specified in a [#[link] attribute](https://doc.rust-lang.org/reference/items/external-blocks.html#the-link-attribute). If the kind is not specified in the link attribute or on the command-line, it will link a dynamic library by default, except when building a static executable. If the kind is specified on the command-line, it will override the kind specified in a link attribute.

In the original code, the kind is not specified in the `link` attribute nor on the command line, so wouldn't cargo try to link a dynamic library?

I understand that the mbedtls-sys build script builds a static lib version of mbedtls and hence we want to link that static lib, but I haven't spotted how the build script tells Cargo that it's a static lib. The documentation (or at least my interpretation of it) seems to indicate that Cargo would try to link a dynamic lib by default, so I would expect the original code to fail. Probably I'm missing something...

Reply:
My guess is Rust tries to link in whatever way it can, it finds that the only available option is static, and does that.

I also note that `mbedtls-platform-support` already does this:
```rust
println!("cargo:rustc-link-lib=static=mbedtls");
println!("cargo:rustc-link-lib=static=mbedx509");
println!("cargo:rustc-link-lib=static=mbedcrypto");
```

Reply:
Anyway, this specific change seems fine.